controllers/actions.github.com/autoscalingrunnerset_controller.go

```diff
@@ -136,6 +137,22 @@ func (r *AutoscalingRunnerSetReconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{}, nil
 	}
 
+	if autoscalingRunnerSet.Labels[LabelKeyKubernetesVersion] != build.Version {
+		if err := r.Delete(ctx, autoscalingRunnerSet); err != nil {
+			log.Error(err, "Failed to delete autoscaling runner set on version mismatch",
+				"targetVersion", build.Version,
+				"observedVersion", autoscalingRunnerSet.Labels[LabelKeyKubernetesVersion],
+			)
+			return ctrl.Result{}, nil
+		}
+
+		log.Info("Successfully issued a delete for autoscaling runner set on version mismatch",
```

Comment:
So this message is correct, but it doesn't really explain why there's a version mismatch in the first place. Maybe we can something like:

> Autoscaling runner set version doesn't match the build version. Deleting the resource.

Reply:
Awesome, I like it